38/ops.html

```diff
@@ -4105,7 +4105,10 @@ <h3>Reverting to ZooKeeper mode During the Migration</h3>
             </li>
             <li>
               Using <code>zookeeper-shell.sh</code>, run <code>rmr /controller</code> so that one
-              of the brokers can become the new old-style controller.
+              of the brokers can become the new old-style controller. Additionally, run
+              <code>get /migration</code> followed by <code>rmr /migration</code> to clear the
+              migration state from ZooKeeper. This will allow you to re-attempt the migration
+              in the future. The data read from "/migration" can be useful for debugging.
```

Comment:
We also need to update it in L4147 (`Migrating brokers to KRaft` row)

Reply:
oh, I don't notice this comment. https://github.com/apache/kafka/pull/16770 is merged. I will sync the changes back to kafka repo if it needs

Reply:
I have commented the corresponding JIRA to understand in which state the user rollbacks the cluster to restart the migration because I tried this scenario more times without deleting the /migration znode and it worked fine.

Reply:
Thanks @showuon, added the second mention. I wonder why we have this duplication? 🤔

Reply:
> I wonder why we have this duplication?

it seems we list all directions for reverting for each migration phase. That is a bit verbose, but they are useful. However, it is weird to use "table format" to display so much texts. Maybe we should to a bit refactor for it. Also, the same direction can have its hyperlink to be shared in different migration phase.

Reply:
I file jira for it (https://issues.apache.org/jira/browse/KAFKA-17271)